### PR TITLE
Dynamically create workflows and tag images

### PR DESCRIPTION
**Pull request by t-young31**
See https://github.com/UCLH-Foundry/FlowEHR/pull/144

### CODE REVIEW

File: .github/workflows/acr_deploy_reusable.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        description: "Token name to login to the ACR"
+        required: true
+      ACR_PASSWORD:
+        description: "Token password to login to the ACR"
+        required: true
+      IMAGE_NAME:
+        description: "Name of the docker image to push to the ACR"
+        required: true
+      ARM_CLIENT_ID:
+        required: false
```
Comment (jjgriff93) [inline .github/workflows/acr_deploy_reusable.yml:23]: Perhaps add descriptions so people know why they're optional and when needed (i.e. deploying to staging slots)

File: .github/workflows/acr_deploy_reusable.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.environment }}
+
+    steps:
+      - name: Set image tag
+        id: set
+        run: |
```
Comment (jjgriff93) [inline .github/workflows/acr_deploy_reusable.yml:41]: Really nice!

File: .github/workflows/slot_swap_reusable.yml [hunk 3]
```diff
@@ -1,14 +1,27 @@
...
+        required: true
+      WEBAPP_ID:
+        required: true
 
 jobs:
   deploy:
     runs-on: ubuntu-latest
-    environment: prod
+    environment: ${{ inputs.environment }}
     steps:
```
Comment (jjgriff93) [inline .github/workflows/slot_swap_reusable.yml:26]:
```suggestion
```

**Review by jjgriff93 — APPROVED** [review]
Great stuff

File: .github/workflows/slot_swap_reusable.yml (hunk 3 above)
Comment (t-young31) [reply .github/workflows/slot_swap_reusable.yml:26]: disagree 😉

File: .github/workflows/acr_deploy_reusable.yml (hunk 1 above)
Comment (t-young31) [reply .github/workflows/acr_deploy_reusable.yml:23]: I thought about this but this shouldn't be something that devs need to understand so didn't bother – just more documentation for me to get wrong. ahaha. we can circle back if needed

**Pull request merged**